utils/auth.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect } from 'react'
+import Router from 'next/router'
+import nextCookie from 'next-cookies'
+import cookie from 'js-cookie'
+
+export const login = ({ token }) => {
+  cookie.set('token', token, { expires: 1 })
+  window.location.replace("/")
+}
+
+export const auth = ctx => {
+  const { token } = nextCookie(ctx)
+
+  // If there's no token, it means the user is not logged in.
+  if (!token) {
+    if (typeof window === 'undefined') {
+      ctx.res.writeHead(302, { Location: '/login' })
+      ctx.res.end()
+    } else {
+      Router.push('/login')
+    }
+  }
+
+  return token
+}
+
+export const logout = () => {
+  cookie.remove('token')
+  // to support logging out from all windows
+  window.localStorage.setItem('logout', Date.now())
+  Router.push('/login')
+}
+
+export const withAuthSync = WrappedComponent => {
```

Comment:
the point of this is to wrap every component that is behind OAuth in this, so a user cannot get to it so you would use this for the profile page